src/api/core.rs

```diff
@@ -202,6 +202,30 @@ fn format_value_array_as_string(v: &[Value]) -> String {
     )
 }
 
+/// `Metadata` uses by `KeyValue` pairs to represent related metadata.
+#[cfg_attr(feature = "serialize", derive(Deserialize, Serialize))]
+#[derive(Clone, Debug, PartialOrd, PartialEq, Default)]
+pub struct Metadata(String);
+
+impl Metadata {
+    /// Return underlying string
+    pub fn as_str(&self) -> &str {
+        self.0.as_str()
+    }
+}
+
+impl From<String> for Metadata {
+    fn from(s: String) -> Metadata {
+        Metadata(s)
```

Comment:
Should trim here I guess?

> Leading and trailing OWS is allowed but MUST be trimmed when converting the header into a data structure.

Reply:
Hmm maybe we should add the trim logic in `TextMapPropagator`.  Also maybe not store the separator(`;`) in metadata. The reason is I think Baggage should be format agnostic, which means we shouldn't introduce format conversion logic inside metadata, those should be in propagator of that format.